lib/base-compiler.js

```diff
@@ -370,6 +371,13 @@ class BaseCompiler {
             });
     }
 
+    processAsm(result, filters) {
```

Comment:
For compilers which do not possibly emit llvm ir, we might want to override this function to only forward its args to `this.asm.process`

Reply:
Okay, but compilers that don't overwrite this should always use `asm.process()`because we explicitly check for LLVM IR output. 
I'm unsure about what exactly I should do about this. :)